### PR TITLE
Support mi supported datatypes

### LINKED ISSUE #22: Support all data types supported by wso2mi
**Description:**
$subject

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Fixes #22

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/mi/Mediator.java [hunk 1]
```diff
@@ -51,13 +59,21 @@ private static String getResultProperty(SimpleMessageContext context) {
...
     public void mediate(SimpleMessageContext context) {
+        String balFunctionReturnType = context.getProperty(Constants.RETURN_TYPE).toString();
         Callback returnCallback = new Callback() {
             public void notifySuccess(Object result) {
                 log.info("Notify Success");
-                OMElement omElement = BXmlConverter.toOMElement((BXml) result);
-                if (omElement != null) {
-                    context.setProperty(getResultProperty(context), BXmlConverter.toOMElement((BXml) result));
+                Object res = result;
+                if (Objects.equals(balFunctionReturnType, "xml")) {
```
Comment (hasithaa) [inline native/src/main/java/io/ballerina/stdlib/mi/Mediator.java]: move these to constants.

File: compiler-plugin/src/main/java/io/ballerina/stdlib/mi/plugin/AnnotationAnalysisTask.java [hunk 2]
```diff
@@ -57,15 +61,36 @@ private static void setParameters(FunctionSymbol functionSymbol, Component compo
 
             for (int i = 0; i < noOfParams; i++) {
                 ParameterSymbol parameterSymbol = parameterSymbols.get(i);
-                if (parameterSymbol.getName().isEmpty()) continue;
-                Param param = new Param(Integer.toString(i), parameterSymbol.getName().get());
-                component.addBalFuncParams(param);
+                Optional<String> optParamName = parameterSymbol.getName();
+                if (optParamName.isEmpty()) continue;
```
Comment (hasithaa) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/mi/plugin/AnnotationAnalysisTask.java]: We don't follow this code style. Also, check check-style is configured.

File: native/src/main/java/io/ballerina/stdlib/mi/Mediator.java (hunk 1 above)
Comment (KavinduZoysa) [reply native/src/main/java/io/ballerina/stdlib/mi/Mediator.java]: Fixed

File: compiler-plugin/src/main/java/io/ballerina/stdlib/mi/plugin/AnnotationAnalysisTask.java (hunk 2 above)
Comment (KavinduZoysa) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/mi/plugin/AnnotationAnalysisTask.java]: Fixed

**Review by hasithaa — APPROVED** [review]

**Pull request merged**